python_modules/dagster/dagster/core/instance/config.py

```diff
@@ -76,6 +101,20 @@ def config_field_for_configurable_class():
     return Field(configurable_class_schema(), is_required=False)
 
 
+def storage_config_schema():
+    return Field(
+        Selector(
+            {
+                "postgres": Field(pg_config()),
```

Comment:
one thing to note here, is that be reusing the `pg_config` schema used for the individual storages, the instance storage config looks like something like this:

```
storage:
  postgres:
    postgres_db:
      username: test
      password: test
      hostname: test
      db_name: test
```

instead of the more terse:
```
storage:
  postgres:
    username: test
    password: test
    hostname: test
    db_name: test
```

The tradeoffs are consistency with the legacy config vs a more succinct config.

Reply:
ya if we want to support direct url StringSource and piece-meal (XXX_db variant), we need another selector layer like this. Its a little clowny but not the worst.

Reply:
I'm not following, what are the XXX_db variants? Could we have the direct url string source be an optional field under 'postgres'? Could `postgres` be a scalar union?